stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/internal/service/CallService.kt

```diff
@@ -164,60 +169,89 @@ internal open class CallService : Service() {
             intent ?: Intent(context, CallService::class.java)
         }
 
-        fun showIncomingCall(context: Context, callId: StreamCallId, callDisplayName: String?, callServiceConfiguration: CallServiceConfig = callServiceConfig()) {
+        fun showIncomingCall(
+            context: Context,
+            callId: StreamCallId,
+            callDisplayName: String?,
+            callServiceConfiguration: CallServiceConfig = callServiceConfig(),
+            notification: Notification?,
+        ) {
             val hasActiveCall = StreamVideo.instanceOrNull()?.state?.activeCall?.value != null
-
-            if (!hasActiveCall) {
-                ContextCompat.startForegroundService(
-                    context,
-                    buildStartIntent(
+            safeCallWithResult {
+                val result = if (!hasActiveCall) {
+                    ContextCompat.startForegroundService(
                         context,
-                        callId,
-                        TRIGGER_INCOMING_CALL,
-                        callDisplayName,
-                        callServiceConfiguration,
-                    ),
-                )
-            } else {
+                        buildStartIntent(
+                            context,
+                            callId,
+                            TRIGGER_INCOMING_CALL,
+                            callDisplayName,
+                            callServiceConfiguration,
+                        ),
+                    )
+                    ComponentName(context, CallService::class.java)
+                } else {
+                    context.startService(
+                        buildStartIntent(
+                            context,
+                            callId,
+                            TRIGGER_INCOMING_CALL,
+                            callDisplayName,
+                            callServiceConfiguration,
+                        ),
+                    )
+                }
+                result!!
+            }.onError {
+                // Show notification
```

Comment:
So when the service cannot be started, we just show the incoming call notification. And the service will be eventually started with the ONGOING trigger when the call is accepted. Is that correct? And what guarantees that the fg service can be started this second time? Can't it fail for a similar reason, like: "if you start the app from a notification, but you go into background while the service is starting" or if we try to start the fg service before the app reaches the foreground?

Reply:
The ONGOING trigger is used when the UI is visible.

Reply:
In the worst case, we'll end up with a running call and a notification, without the fg service.

Reply:
Not ideal, but at least we're not crashing anymore in those edge-cases. Sounds good.